tests/pkg/cspc/cspcspecbuilder/specbuilder.go

```diff
@@ -0,0 +1,223 @@
+	cd.UnUsedDisk[diskName] = true
+	delete(cd.UsedDisks, diskName)
+}
+
+func NewCSPCSpecBuilder(cspcCache *cspccache.CSPCResourceCache, infra *infra.Infrastructure) *CSPCSpecBuilder {
+	// Initialize CSPCSpecData
+	cspcSpecData := NewCSPCSpecData()
+	for _, nodeName := range cspcCache.NodeList {
+		cspcSpecData.UnUsedNodes[nodeName] = true
+		for _, bd := range cspcCache.NodeDisk[nodeName] {
+			cspcSpecData.UnUsedDisk[bd] = true
+		}
+	}
+	return &CSPCSpecBuilder{
+		CSPCCache:    cspcCache,
+		CSPCSpecData: cspcSpecData,
+		Infra:        infra,
+	}
+}
+
+// ToDo: Complete following functions
+
+func (c *CSPCSpecBuilder) ReplaceBlockDevice(oldBD, replaceWith string) {
+
+}
+
+func (c *CSPCSpecBuilder) AddRaidGroup(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) RemoveRaidGroup(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddResourceLimit(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddDefaultResourceLimit() {
+
+}
+
+func (c *CSPCSpecBuilder) AddTolerations(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddDefaultTolerations() {
+
+}
+
+func (c *CSPCSpecBuilder) AddPriorityClass(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddDefaultPriorityClass() {
+
+}
+
+func (c *CSPCSpecBuilder) AddCompression(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddDefaultCompression() {
+
+}
+
+func (c *CSPCSpecBuilder) AddROThreshold(location int) {
+
+}
+
+func (c *CSPCSpecBuilder) AddDefaultROThreshold() {
+
+}
+
+// RemovePoolSpec removes a pool spec from the CSPC.
+func (c *CSPCSpecBuilder) RemovePoolSpec() *CSPCSpecBuilder {
+	currentSize := len(c.CSPC.Spec.Pools)
+	c.CSPC.Spec.Pools = c.CSPC.Spec.Pools[:currentSize-1]
+	return c
+}
+
+// AddPoolSpec adds a pool spec in the CSPC in accordance with the provided
+// arguments
+func (c *CSPCSpecBuilder) AddPoolSpec(nodeName string, poolType string, bdCount int) {
+	c.CSPCSpecData.AddNodeToUsedSet(nodeName)
+	nodeSelector := c.CSPCCache.Nodes[nodeName]
+	blockDevices := make([]cstor.CStorPoolInstanceBlockDevice, 0)
+
+	if len(c.CSPCCache.NodeDisk[nodeName]) < bdCount {
+		klog.Fatalf("Not enough block "+
+			"devices available for node %s: want %d,got %d",
+			nodeName, bdCount, c.CSPCCache.NodeDisk[nodeName])
+	}
+
+	for i := 0; i < bdCount; i++ {
+		newBlockDevice := cstor.CStorPoolInstanceBlockDevice{
+			BlockDeviceName: c.CSPCCache.NodeDisk[nodeName][i],
+		}
+		blockDevices = append(blockDevices, newBlockDevice)
+		c.CSPCSpecData.AddDiskToUsedSet(newBlockDevice.BlockDeviceName)
+	}
+
+	newPoolSpec := cstor.NewPoolSpec().
+		WithNodeSelector(nodeSelector).
+		WithDataRaidGroups(
+			*cstor.NewRaidGroup().
+				WithCStorPoolInstanceBlockDevices(
+					blockDevices...,
+				),
+		).
+		WithPoolConfig(*cstor.NewPoolConfig().WithDataRaidGroupType(poolType))
+
+	c.CSPC.WithPoolSpecs([]cstor.PoolSpec{*newPoolSpec}...)
+}
+
+func (c *CSPCSpecBuilder) BuildCSPC(poolType string, bdCount int) *CSPCSpecBuilder {
+	nodeCount := c.Infra.NodeCount
+	if nodeCount == 1 {
```

Comment:
Instead of segregating based on node count. Can we pass pool count as an argument to  `BuildNodeCSPCSpec` function? This pool count we can take it from `ginkgo flag` instead of taking `node count`

Reply:
node count is coming from a ginkgo flag only.
https://github.com/openebs/cstor-operators/pull/118/files#diff-3dabef26c55a41523b25b5058bb2d61eR29
Is this what you mean?

Reply:
Yes!! Let's change that variable from `nodeCount` to `poolSpecCount` (or) `poolCount`. Having `nodeCount` will miss lead. Example: I have `n` node cluster and among them, I have `m` storage nodes where the disks were attached(where m < n) in this configuration I want to create a `p` pools(where p < m). Now saying nodeCount as flag will lead to confusion

Reply:
Check this https://github.com/openebs/cstor-operators/pull/118#discussion_r451983155

nodecount simply denotes -- nodes where pools can be provisioned. 
In the test you can decide how many pools we want. 
IMO let us not tie this --> poolCount == soomethingElseCount  -- NOw somethingElse necessarily not be the pool key word.